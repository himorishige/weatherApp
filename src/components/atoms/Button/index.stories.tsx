import React from 'react';
import { Meta, Story } from '@storybook/react';
import Button from '.';

export default {
  title: 'WeatherApp/Atoms/Button',
  component: Button,
} as Meta;

const Template: Story = (args) => <Button {...args} />;

export const SearchButton = Template.bind({});
