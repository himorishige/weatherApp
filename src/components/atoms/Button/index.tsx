/** @jsxImportSource @emotion/react */
import React, { memo } from 'react';
import { css } from '@emotion/react';

export type Props = {
  disabled?: boolean;
  clickHandler?: () => void;
};

const Button: React.VFC<Props> = memo(({ disabled = false, clickHandler }) => {
  return (
    <button type="button" css={buttonStyle} disabled={disabled} onClick={clickHandler}>
      検索する
    </button>
  );
});

export default Button;

const buttonStyle = css`
  padding: 0.5rem 1rem;
  text-align: center;
  color: var(--white-color);
  background: var(--primary-color);
  cursor: pointer;
  border: none;
  border-radius: 4px;
  width: 100%;
  transition: opacity 0.3s;
  &:hover {
    opacity: 0.7;
  }
  &:disabled {
    background: #cccccc;
    cursor: not-allowed;
  }
`;
