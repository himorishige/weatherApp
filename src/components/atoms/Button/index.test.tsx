import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '.';

describe('Atoms/Button', () => {
  it('正しくレンダリングされている', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
