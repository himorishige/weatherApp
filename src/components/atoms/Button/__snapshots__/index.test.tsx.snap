// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button 正しくレンダリングされている 1`] = `
<button
  className="css-vzmsot"
  disabled={false}
  type="button"
>
  検索する
</button>
`;
